api/src/main/java/com/datastrato/gravitino/rel/partitions/Partitions.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.rel.partitions;
+
+import com.datastrato.gravitino.rel.expressions.literals.Literal;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Objects;
+
+/** The helper class for partition expressions. */
+public class Partitions {
+
+  /**
+   * Creates a range partition.
+   *
+   * @param name The name of the partition.
+   * @param upper The upper bound of the partition.
+   * @param lower The lower bound of the partition.
+   * @param properties The properties of the partition.
+   * @return The created partition.
+   */
+  public static Partition range(
+      String name, Literal<?> upper, Literal<?> lower, Map<String, String> properties) {
+    return new RangePartitionImpl(name, upper, lower, properties);
+  }
+
+  /**
+   * Creates a list partition.
+   *
+   * @param name The name of the partition.
+   * @param lists The values of the list partition.
+   * @param properties The properties of the partition.
+   * @return The created partition.
+   */
+  public static Partition list(String name, Literal<?>[][] lists, Map<String, String> properties) {
+    return new ListPartitionImpl(name, lists, properties);
+  }
+
+  /**
+   * Creates an identity partition.
+   *
+   * @param name The name of the partition.
+   * @param fieldNames The field names of the identity partition.
+   * @param value The value of the identity partition.
+   * @param properties The properties of the partition.
+   * @return The created partition.
+   */
+  public static Partition identity(
```

Comment:
Since here we have the ordering requirement, I would suggest you add more comments here.